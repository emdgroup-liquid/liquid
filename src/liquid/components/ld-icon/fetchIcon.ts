import { getAssetPath } from '@stencil/core'

const iconCache = {}
const requestCache = {}

export async function fetchIcon(icon: string): Promise<string> {
  if (iconCache[icon]) {
    return iconCache[icon]
  }
  if (!requestCache[icon]) {
    requestCache[icon] = fetch(getAssetPath(`./assets/${icon}.svg`))
      .then((resp) => resp.text())
      .catch((err) => {
        console.error(`"${icon}" is not a valid name`, err)
        return ''
      })
  }

  const path = await requestCache[icon]
  iconCache[icon] = path

  return path
}
