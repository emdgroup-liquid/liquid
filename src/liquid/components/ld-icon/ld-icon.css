:host {
  display: inline-flex;

  > div {
    display: inline-flex;
  }
}
