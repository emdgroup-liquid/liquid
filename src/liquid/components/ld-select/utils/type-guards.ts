export const isLdOption = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptionElement | HTMLLdOptionInternalElement =>
  ["LD-OPTION", "LD-OPTION-INTERNAL"].includes((el as HTMLElement)?.tagName);

export const isLdOptgroup = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptgroupElement | HTMLLdOptgroupInternalElement =>
  ["LD-OPTGROUP", "LD-OPTGROUP-INTERNAL"].includes(
    (el as HTMLElement)?.tagName,
  );

export const isLdOptionInternal = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptionInternalElement =>
  ["LD-OPTION-INTERNAL"].includes((el as HTMLElement)?.tagName);

export const isLdOptgroupInternal = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptgroupInternalElement =>
  ["LD-OPTGROUP-INTERNAL"].includes((el as HTMLElement)?.tagName);

type HTMLLdOptInternal =
  | HTMLLdOptionInternalElement
  | HTMLLdOptgroupInternalElement;
export const isLdOptInternalHidden = (
  opt: HTMLLdOptInternal,
): opt is
  | (HTMLLdOptInternal & {
      hidden: true;
    })
  | (HTMLLdOptInternal & {
      filtered: true;
    }) => {
  return opt.hidden || opt.filtered;
};
