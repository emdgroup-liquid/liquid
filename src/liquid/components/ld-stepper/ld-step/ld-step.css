.ld-step {
  --ld-step-description-font-size: var(--ld-step-description-font-size-md);
  --ld-step-description-font-size-sm: 0.625rem;
  --ld-step-description-font-size-md: 0.75rem;
  --ld-step-description-font-size-lg: 0.875rem;
  --ld-step-dot-offset: calc(
    (var(--ld-step-dot-size-with-icon) - var(--ld-step-dot-size)) / 2
  );
  --ld-step-dot-size: var(--ld-step-dot-size-default);
  --ld-step-dot-size-default: var(--ld-step-dot-size-md);
  --ld-step-dot-size-sm: var(--ld-sp-12);
  --ld-step-dot-size-md: var(--ld-sp-16);
  --ld-step-dot-size-lg: 1.25rem;
  --ld-step-dot-size-with-icon: var(--ld-step-dot-size-with-icon-md);
  --ld-step-dot-size-with-icon-sm: 1.25rem;
  --ld-step-dot-size-with-icon-md: var(--ld-sp-24);
  --ld-step-dot-size-with-icon-lg: var(--ld-sp-32);
  --ld-step-icon-size: calc(var(--ld-step-dot-size-with-icon) * 0.5);
  --ld-step-icon-size-custom: calc(var(--ld-step-dot-size-with-icon) * 0.6);
  --ld-step-icon-offset: calc(
    (var(--ld-step-dot-size-with-icon) - var(--ld-step-icon-size)) / 2
  );
  --ld-step-label-font-size: var(--ld-step-label-font-size-md);
  --ld-step-label-font-size-sm: 0.75rem;
  --ld-step-label-font-size-md: 0.875rem;
  --ld-step-label-font-size-lg: 1.125rem;
  --ld-step-line-thickness: var(--ld-step-line-thickness-md);
  --ld-step-line-thickness-sm: var(--ld-sp-2);
  --ld-step-line-thickness-md: 0.1875rem;
  --ld-step-line-thickness-lg: var(--ld-sp-4);
  --ld-step-text-distance: var(--ld-sp-16);
  --ld-step-text-margin: calc(
    var(--ld-step-dot-size-with-icon) + var(--ld-step-text-distance)
  );
  --ld-step-vertical-label-margin-top: calc(
    /* remove actual line-height of the label from dot size to get the difference */
      (var(--ld-step-dot-size-with-icon) - var(--ld-step-label-font-size) * 1.4) /
      2
  );
  --ld-step-vertical-text-gap: var(--ld-step-vertical-text-gap-md);
  --ld-step-vertical-text-gap-sm: var(--ld-sp-4);
  --ld-step-vertical-text-gap-md: var(--ld-sp-6);
  --ld-step-vertical-text-gap-lg: var(--ld-sp-8);

  /* colors */
  --ld-step-bg-col: var(--ld-thm-primary);
  --ld-step-col: var(--ld-step-col-default);
  --ld-step-col-default: var(--ld-col-neutral-400);
  --ld-step-col-done: var(--ld-thm-primary);
  --ld-step-dot-col: var(--ld-step-dot-col-default);
  --ld-step-dot-col-default: var(--ld-col-neutral-200);
  --ld-step-dot-col-current: var(--ld-thm-secondary);
  --ld-step-dot-col-done: var(--ld-thm-primary);
  --ld-step-description-col: var(--ld-step-description-col-default);
  --ld-step-description-col-default: var(--ld-step-col);
  --ld-step-description-col-done: var(--ld-col-neutral-900);
  --ld-step-icon-col: var(--ld-col-wht);
  --ld-step-line-start-col: var(--ld-step-line-col-default);
  --ld-step-line-end-col: var(--ld-step-line-col-default);
  --ld-step-line-col-default: var(--ld-col-neutral-200);
  --ld-step-line-col-done: var(--ld-thm-primary);

  /* on-brand colors */
  --ld-step-brand-col: var(--ld-col-wht-alpha-highest);
  --ld-step-brand-col-done: var(--ld-col-wht);
  --ld-step-description-brand-col: var(--ld-col-wht-alpha-highest);
  --ld-step-description-brand-col-done: var(--ld-col-wht);
  --ld-step-dot-brand-col: var(--ld-thm-primary-focus);
  --ld-step-dot-brand-col-done: var(--ld-col-wht);
  --ld-step-brand-icon-col: var(--ld-thm-primary);
  --ld-step-line-brand-col: var(--ld-thm-primary-focus);
  --ld-step-line-brand-col-done: var(--ld-col-wht);

  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  position: relative;

  a,
  button {
    align-items: center;
    background: transparent;
    border: 0;
    color: var(--ld-step-col);
    display: flex;
    flex-direction: column;
    font-family: var(--ld-font-body);
    font-size: var(--ld-step-label-font-size);
    font-weight: 700;
    line-height: 140%;
    margin: var(--ld-step-text-margin) 0 0; /* Safari reset */
    padding: 0;
    text-align: center;

    &::before {
      background: var(--ld-step-dot-col);
      border-radius: var(--ld-br-full);
      box-sizing: border-box;
      content: '';
      display: block;
      height: var(--ld-step-dot-size);
      position: absolute;
      top: var(--ld-step-dot-offset);
      width: var(--ld-step-dot-size);
      z-index: 1;
    }
  }

  .ld-icon {
    --ld-icon-size-sm: var(--ld-step-icon-size);
    --ld-icon-size-md: var(--ld-step-icon-size);
    --ld-icon-size-lg: var(--ld-step-icon-size);
    color: var(--ld-step-icon-col);
    pointer-events: none;
    position: absolute;
    top: var(--ld-step-icon-offset);
    z-index: 1;
  }

  &--brand-color {
    --ld-step-col-default: var(--ld-step-brand-col);
    --ld-step-col-done: var(--ld-step-brand-col-done);
    --ld-step-description-col-default: var(--ld-step-description-brand-col);
    --ld-step-description-col-done: var(--ld-step-description-brand-col-done);
    --ld-step-dot-col-default: var(--ld-step-dot-brand-col);
    --ld-step-dot-col-done: var(--ld-step-dot-brand-col-done);
    --ld-step-icon-col: var(--ld-step-brand-icon-col);
    --ld-step-line-col-default: var(--ld-step-line-brand-col);
    --ld-step-line-col-done: var(--ld-step-line-brand-col-done);
  }

  &--custom-icon {
    --ld-step-icon-size: var(--ld-step-icon-size-custom);
  }

  &--with-icon {
    --ld-step-dot-size: var(--ld-step-dot-size-with-icon);
  }

  &--current {
    --ld-step-col: var(--ld-step-col-done);
    --ld-step-description-col: var(--ld-step-description-col-done);
    --ld-step-dot-col: var(--ld-step-dot-col-current);
    --ld-step-line-start-col: var(--ld-step-line-col-done);
  }

  &--done {
    --ld-step-col: var(--ld-step-col-done);
    --ld-step-description-col: var(--ld-step-description-col-done);
    --ld-step-dot-col: var(--ld-step-dot-col-done);
    --ld-step-line-start-col: var(--ld-step-line-col-done);
    --ld-step-line-end-col: var(--ld-step-line-col-done);

    a,
    button {
      cursor: pointer;
    }
  }

  &--optional {
    --ld-step-icon-col: var(--ld-step-dot-col);

    a,
    button {
      &::before {
        background: transparent;
        border: solid var(--ld-step-line-thickness) var(--ld-step-dot-col);
      }
    }
  }

  &--skipped {
    --ld-step-col: var(--ld-step-col-done);
    --ld-step-description-col: var(--ld-step-description-col-done);
    --ld-step-dot-col: var(--ld-step-dot-col-done);
    --ld-step-line-start-col: var(--ld-step-line-col-done);
    --ld-step-line-end-col: var(--ld-step-line-col-done);

    a,
    button {
      cursor: pointer;
    }
  }

  &--vertical {
    align-items: start;
    gap: var(--ld-step-vertical-text-gap);
    padding: 0;
    text-align: left;

    a,
    button {
      margin: var(--ld-step-vertical-label-margin-top) 0 0
        var(--ld-step-text-margin);

      &::before {
        flex-shrink: 0;
        left: calc(var(--ld-step-dot-size-with-icon) / 2);
        transform: translateX(-50%);
      }
    }

    .ld-icon {
      left: var(--ld-step-icon-offset);
    }
  }

  :host(:empty) & a,
  :host(:empty) & button,
  a:empty,
  button:empty {
    margin: var(--ld-step-dot-offset) 0 0;

    &,
    &::before {
      position: relative;
      top: 0;
    }
  }

  &--sm {
    --ld-step-description-font-size: var(--ld-step-description-font-size-sm);
    --ld-step-dot-size-default: var(--ld-step-dot-size-sm);
    --ld-step-dot-size-with-icon: var(--ld-step-dot-size-with-icon-sm);
    --ld-step-label-font-size: var(--ld-step-label-font-size-sm);
    --ld-step-line-thickness: var(--ld-step-line-thickness-sm);
    --ld-step-vertical-text-gap: var(--ld-step-vertical-text-gap-sm);
  }

  &--lg {
    --ld-step-description-font-size: var(--ld-step-description-font-size-lg);
    --ld-step-dot-size-default: var(--ld-step-dot-size-lg);
    --ld-step-dot-size-with-icon: var(--ld-step-dot-size-with-icon-lg);
    --ld-step-label-font-size: var(--ld-step-label-font-size-lg);
    --ld-step-line-thickness: var(--ld-step-line-thickness-lg);
    --ld-step-vertical-text-gap: var(--ld-step-vertical-text-gap-lg);
  }
}

:host(:not(:first-of-type)) .ld-step,
.ld-step:not(:first-of-type) {
  &::before {
    content: '';
    background: var(--ld-step-line-start-col);
    height: var(--ld-step-line-thickness);
    left: calc(var(--ld-stepper-gap) / -2);
    position: absolute;
    right: calc(50% + var(--ld-step-dot-size) / 2 - var(--ld-sp-1));
    top: calc(var(--ld-step-dot-size-with-icon) / 2);
    transform: translateY(-50%);
  }

  &.ld-step--vertical::before {
    bottom: calc(100% - var(--ld-step-dot-offset) - var(--ld-sp-1));
    height: auto;
    left: calc(var(--ld-step-dot-size-with-icon) / 2);
    right: auto;
    top: calc(var(--ld-stepper-gap) / -2);
    transform: translatex(-50%);
    width: var(--ld-step-line-thickness);
  }
}

:host(:not(:last-of-type)) .ld-step,
.ld-step:not(:last-of-type) {
  &::after {
    content: '';
    background: var(--ld-step-line-end-col);
    height: var(--ld-step-line-thickness);
    left: calc(50% + var(--ld-step-dot-size) / 2 - var(--ld-sp-1));
    position: absolute;
    right: calc(var(--ld-stepper-gap) / -2);
    top: calc(var(--ld-step-dot-size-with-icon) / 2);
    transform: translateY(-50%);
    z-index: -1;
  }

  &.ld-step--vertical::after {
    bottom: calc(var(--ld-stepper-gap) / -2);
    height: auto;
    left: calc(var(--ld-step-dot-size-with-icon) / 2);
    right: auto;
    top: calc(
      var(--ld-step-dot-size) + var(--ld-step-dot-offset) - var(--ld-sp-1)
    );
    transform: translatex(-50%);
    width: var(--ld-step-line-thickness);
  }
}

.ld-step__description {
  color: var(--ld-step-description-col);
  font-family: var(--ld-font-body);
  font-size: var(--ld-step-description-font-size);
  line-height: 140%;
  margin-left: var(--ld-step-text-margin);
}
