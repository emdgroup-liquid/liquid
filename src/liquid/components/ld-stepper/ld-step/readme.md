---
eleventyNavigation:
  key: Step
  parent: Stepper
layout: layout.njk
title: Step
permalink: components/ld-stepper/ld-step/
---

# ld-step

The `ld-step` component is a subcomponent for `ld-stepper`.

Please refer to the [`ld-stepper` documentation](components/ld-stepper) for usage examples.

---


<!-- Auto Generated Below -->


## Properties

| Property           | Attribute            | Description                                                                         | Type               | Default          |
| ------------------ | -------------------- | ----------------------------------------------------------------------------------- | ------------------ | ---------------- |
| `brandColor`       | `brand-color`        | Switch colors for brand background                                                  | `boolean`          | `false`          |
| `current`          | `current`            | Step is the current step                                                            | `boolean`          | `false`          |
| `description`      | `description`        | Description text to display below the step name (vertical mode only)                | `string`           | `undefined`      |
| `done`             | `done`               | Step is done                                                                        | `boolean`          | `false`          |
| `icon`             | `icon`               | Permanently show a custom icon inside the dot                                       | `string`           | `undefined`      |
| `key`              | `key`                | for tracking the node's identity when working with lists                            | `string \| number` | `undefined`      |
| `labelCurrent`     | `label-current`      | Label for current step (scree-reader only)                                          | `string`           | `'Current'`      |
| `labelDone`        | `label-done`         | Label for step that is done (scree-reader only)                                     | `string`           | `'Done'`         |
| `labelOptional`    | `label-optional`     | Label for step that is optional (scree-reader only)                                 | `string`           | `'Optional'`     |
| `labelSkipped`     | `label-skipped`      | Label for step that was skipped (scree-reader only)                                 | `string`           | `'Skipped'`      |
| `labelWasOptional` | `label-was-optional` | Additional hint in label for step that is done and was optional (scree-reader only) | `string`           | `'was optional'` |
| `ldTabindex`       | `ld-tabindex`        | Tab index of the step                                                               | `number`           | `undefined`      |
| `optional`         | `optional`           | Step may be skipped                                                                 | `boolean`          | `false`          |
| `ref`              | `ref`                | reference to component                                                              | `any`              | `undefined`      |
| `size`             | `size`               | Step size                                                                           | `"lg" \| "sm"`     | `undefined`      |
| `skipped`          | `skipped`            | Step was skipped                                                                    | `boolean`          | `false`          |
| `vertical`         | `vertical`           | Vertical layout                                                                     | `boolean`          | `false`          |


## Events

| Event            | Description | Type                                             |
| ---------------- | ----------- | ------------------------------------------------ |
| `ldstepselected` |             | `CustomEvent<{ index: number; label: string; }>` |


## Methods

### `focusInner() => Promise<void>`

Sets focus on the step

#### Returns

Type: `Promise<void>`




## Shadow Parts

| Part            | Description                                  |
| --------------- | -------------------------------------------- |
| `"button"`      | actual `button` element                      |
| `"description"` | `span` element wrapping the description text |
| `"focusable"`   |                                              |
| `"li"`          | actual `li` element                          |


## Dependencies

### Depends on

- [ld-sr-only](../../ld-sr-only)
- [ld-icon](../../ld-icon)

### Graph
```mermaid
graph TD;
  ld-step --> ld-sr-only
  ld-step --> ld-icon
  style ld-step fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

 
