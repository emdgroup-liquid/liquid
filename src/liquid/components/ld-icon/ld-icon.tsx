import { Build, Component, Host, h, Prop, Watch, Element } from '@stencil/core'
import { getClassNames } from '../../utils/getClassNames'
import { fetchIcon } from './fetchIcon'

/**
 * @slot - (optional) Custom SVG icon (only valid without name prop).
 * @virtualProp ref - reference to component
 * @virtualProp {string | number} key - for tracking the node's identity when working with lists
 * @part icon - Actual SVG element
 */
@Component({
  assetsDirs: ['assets'],
  tag: 'ld-icon',
  styleUrl: 'ld-icon.css',
  shadow: true,
})
export class LdIcon {
  @Element() element: HTMLElement

  /** The icon name. */
  @Prop() name: string = null

  /** Size of the icon. */
  @Prop() size?: 'sm' | 'lg'

  @Watch('name')
  private async loadIconPathData(): Promise<void> {
    if ((!Build.isBrowser && !Build.isTesting) || !this.name) {
      return
    }

    const div = document.createElement('div')
    const iconString = await fetchIcon(this.name)

    div.innerHTML = iconString.replace(
      '<svg',
      '<svg class="ld-icon__svg" part="icon"'
    )
    Array.from(this.element.shadowRoot.children).forEach((child) => {
      if (child.tagName !== 'STYLE') {
        this.element.shadowRoot.removeChild(child)
      }
    })
    this.element.shadowRoot.appendChild(div.firstChild)
  }

  async componentWillLoad(): Promise<void> {
    this.loadIconPathData()
  }

  render() {
    return (
      <Host
        class={getClassNames(['ld-icon', this.size && `ld-icon--${this.size}`])}
        role="presentation"
      >
        {!this.name && <slot></slot>}
      </Host>
    )
  }
}
