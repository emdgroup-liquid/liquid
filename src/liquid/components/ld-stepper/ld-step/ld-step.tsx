import {
  Component,
  Host,
  h,
  Prop,
  Element,
  Event,
  EventEmitter,
  Method,
} from '@stencil/core'
import { getClassNames } from 'src/liquid/utils/getClassNames'

export type SelectedDetail = { index: number; label: string }

/**
 * @virtualProp ref - reference to component
 * @virtualProp {string | number} key - for tracking the node's identity when working with lists
 * @part description - `span` element wrapping the description text
 * @part button - actual `button` element
 * @part li - actual `li` element
 */
@Component({
  tag: 'ld-step',
  styleUrl: 'ld-step.css',
  shadow: true,
})
export class LdStep implements InnerFocusable {
  @Element() el: HTMLLdStepElement
  private button: HTMLButtonElement

  /** Switch colors for brand background */
  @Prop() brandColor = false
  /** Step is the current step */
  @Prop() current = false
  /** Description text to display below the step name (vertical mode only) */
  @Prop() description: string
  /** Step is done */
  @Prop() done = false
  /** Permanently show a custom icon inside the dot */
  @Prop() icon?: HTMLLdIconElement['name']
  /** Tab index of the step */
  @Prop() ldTabindex: number | undefined
  /** Step may be skipped */
  @Prop() optional = false
  /** Step was skipped */
  @Prop() skipped = false
  /** Vertical layout */
  @Prop() vertical = false

  @Event() ldstepselected: EventEmitter<SelectedDetail>

  /** Sets focus on the step */
  @Method()
  async focusInner() {
    this.button?.focus()
  }

  handleClick = () => {
    this.ldstepselected.emit({
      index: Array.from(this.el.parentElement.children).findIndex(
        (child) => child === this.el
      ),
      label: this.el.textContent,
    })
  }

  componentDidLoad() {
    if (this.current) {
      this.handleClick()
    }
  }

  render() {
    return (
      <Host>
        <li
          class={getClassNames([
            'ld-step',
            this.brandColor && 'ld-step--brand-color',
            this.current && 'ld-step--current',
            this.done && 'ld-step--done',
            this.icon && 'ld-step--custom-icon',
            (this.done || this.icon) && 'ld-step--with-icon',
            this.optional && 'ld-step--optional',
            this.skipped && 'ld-step--skipped',
            this.vertical && 'ld-step--vertical',
          ])}
          part="li"
          role="listitem"
        >
          <button
            aria-current={this.current ? 'step' : undefined}
            disabled={!this.done && !this.skipped}
            onClick={this.handleClick}
            part="button focusable"
            ref={(ref) => (this.button = ref)}
            tabIndex={this.ldTabindex}
            type="button"
          >
            <slot></slot>
          </button>
          {(this.done || this.icon) && (
            <ld-icon name={this.icon ?? 'checkmark'} />
          )}
          {this.description && this.vertical && (
            <span class="ld-step__description" part="description">
              {this.description}
            </span>
          )}
        </li>
      </Host>
    )
  }
}
