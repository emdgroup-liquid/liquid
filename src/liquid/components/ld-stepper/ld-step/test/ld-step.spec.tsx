import { h } from '@stencil/core'
import { newSpecPage } from '@stencil/core/testing'
import { LdIcon } from '../../../ld-icon/ld-icon'
import { LdStep } from '../ld-step'
import '../../../../utils/mutationObserver'

describe('ld-step', () => {
  describe('renders', () => {
    it('default', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('with size prop', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step size="lg" />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('with text', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step>Step 1</ld-step>,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('with description', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step description="Example description" />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as current', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step current />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as disabled', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step disabled />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as last active', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step last-active />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as next', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step next />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as done', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step done />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as optional', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step optional />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as skipped', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step optional skipped />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as done but optional', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step done optional />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('on brand color', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step brandColor />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('vertical', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step vertical />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as anchor', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step href="#" />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('as disabled anchor', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step disabled href="#" />,
      })
      expect(page.root).toMatchSnapshot()
    })

    describe('with custom label', () => {
      it('current', async () => {
        const page = await newSpecPage({
          components: [LdStep],
          template: () => <ld-step current labelCurrent="Aktiv" />,
        })
        expect(page.root.shadowRoot.querySelector('ld-sr-only')).toEqualText(
          'Aktiv: '
        )
      })

      it('done', async () => {
        const page = await newSpecPage({
          components: [LdStep],
          template: () => <ld-step done labelDone="Erledigt" />,
        })
        expect(page.root.shadowRoot.querySelector('ld-sr-only')).toEqualText(
          'Erledigt: '
        )
      })

      it('optional', async () => {
        const page = await newSpecPage({
          components: [LdStep],
          template: () => <ld-step optional labelOptional="Skippable" />,
        })
        expect(page.root.shadowRoot.querySelector('ld-sr-only')).toEqualText(
          'Skippable: '
        )
      })

      it('skipped', async () => {
        const page = await newSpecPage({
          components: [LdStep],
          template: () => (
            <ld-step optional skipped labelSkipped="Übersprungen" />
          ),
        })
        expect(page.root.shadowRoot.querySelector('ld-sr-only')).toEqualText(
          'Übersprungen: '
        )
      })

      it('was optional', async () => {
        const page = await newSpecPage({
          components: [LdStep],
          template: () => (
            <ld-step
              done
              optional
              labelDone="Erledigt"
              labelWasOptional="war optional"
            />
          ),
        })
        expect(page.root.shadowRoot.querySelector('ld-sr-only')).toEqualText(
          'Erledigt (war optional): '
        )
      })
    })

    it('with custom icon', async () => {
      const page = await newSpecPage({
        components: [LdIcon, LdStep],
        template: () => <ld-step icon="favorite" />,
      })
      expect(page.root).toMatchSnapshot()
    })

    it('with ldTabindex', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step ldTabindex={5} />,
      })
      expect(page.root.shadowRoot.querySelector('button').tabIndex).toBe(5)
    })

    it('with cloned attribute', async () => {
      const page = await newSpecPage({
        components: [LdStep],
        template: () => <ld-step aria-hidden="true" />,
      })
      await page.waitForChanges()
      // Test environment does not seem to support the `ariaHidden` prop, here
      expect(
        page.root.shadowRoot.querySelector('button').getAttribute('aria-hidden')
      ).toBe('true')
    })
  })

  it('initially emits ldstepselected event when current', async () => {
    const page = await newSpecPage({
      components: [LdStep],
      template: () => <ld-step current>Step 1</ld-step>,
    })
    const eventHandler = jest.fn()

    page.root.addEventListener('ldstepselected', eventHandler)
    page.rootInstance.componentDidLoad()

    expect(eventHandler).toHaveBeenCalledTimes(1)
    expect(eventHandler).toHaveBeenCalledWith(
      expect.objectContaining({
        detail: {
          index: 0,
          label: 'Step 1',
        },
      })
    )
  })

  it('emits ldstepselected event on click', async () => {
    const page = await newSpecPage({
      components: [LdStep],
      template: () => <ld-step done>Step 1</ld-step>,
    })
    const eventHandler = jest.fn()
    const button = page.root.shadowRoot.querySelector('button')

    page.root.addEventListener('ldstepselected', eventHandler)
    button.click()

    expect(eventHandler).toHaveBeenCalledWith(
      expect.objectContaining({
        detail: {
          index: 0,
          label: 'Step 1',
        },
      })
    )
  })

  it('does not emit ldstepselected event on click when current', async () => {
    const page = await newSpecPage({
      components: [LdStep],
      template: () => <ld-step current>Step 1</ld-step>,
    })
    const eventHandler = jest.fn()
    const button = page.root.shadowRoot.querySelector('button')

    page.root.addEventListener('ldstepselected', eventHandler)
    button.click()

    expect(eventHandler).not.toHaveBeenCalledWith()
  })

  it('does not emit ldstepselected event on click when disabled', async () => {
    const page = await newSpecPage({
      components: [LdStep],
      template: () => <ld-step disabled>Step 1</ld-step>,
    })
    const eventHandler = jest.fn()
    const button = page.root.shadowRoot.querySelector('button')

    page.root.addEventListener('ldstepselected', eventHandler)
    button.click()

    expect(eventHandler).not.toHaveBeenCalledWith()
  })

  it('allows to set inner focus', async () => {
    const page = await newSpecPage({
      components: [LdStep],
      template: () => <ld-step done>Step 1</ld-step>,
    })
    const button = page.root.shadowRoot.querySelector('button')

    button.focus = jest.fn()
    await page.root.focusInner()

    expect(button.focus).toHaveBeenCalled()
  })
})
