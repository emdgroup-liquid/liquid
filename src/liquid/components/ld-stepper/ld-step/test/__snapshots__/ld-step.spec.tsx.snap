// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ld-step renders as current 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--current" part="li" role="listitem">
      <ld-sr-only>
        Current:
      </ld-sr-only>
      <button aria-current="step" disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders as done 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--done ld-step--with-icon" part="li" role="listitem">
      <ld-sr-only>
        Done:
      </ld-sr-only>
      <button part="button focusable" type="button">
        <slot></slot>
      </button>
      <ld-icon name="checkmark"></ld-icon>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders as done but optional 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--done ld-step--optional ld-step--with-icon" part="li" role="listitem">
      <ld-sr-only>
        Done (was optional):
      </ld-sr-only>
      <button part="button focusable" type="button">
        <slot></slot>
      </button>
      <ld-icon name="checkmark"></ld-icon>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders as optional 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--optional" part="li" role="listitem">
      <ld-sr-only>
        Optional:
      </ld-sr-only>
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders as skipped 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--optional ld-step--skipped" part="li" role="listitem">
      <ld-sr-only>
        Skipped:
      </ld-sr-only>
      <button part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders default 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders on brand color 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--brand-color" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders vertical 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--vertical" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders with custom icon 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--custom-icon ld-step--with-icon" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
      <ld-icon class="ld-icon" role="presentation">
        <mock:shadow-root>
          Not Found
        </mock:shadow-root>
      </ld-icon>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders with description 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders with size prop 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step ld-step--lg" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
</ld-step>
`;

exports[`ld-step renders with text 1`] = `
<ld-step>
  <mock:shadow-root>
    <li class="ld-step" part="li" role="listitem">
      <button disabled="" part="button focusable" type="button">
        <slot></slot>
      </button>
    </li>
  </mock:shadow-root>
  Step 1
</ld-step>
`;
